src/components/PokemonList/PokemonList.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {useState, useEffect} from 'react';
+
+import {getList, getOne} from '../../api/pokemons.api';
+import PokemonCard from '../PokemonCard/PokemonCard';
+
+import styles from './PokemonList.module.css'
+
+interface Pokemon {
+    name: string;
+    url: string;
+    sprites: { front_default: string };
+}
+
+interface PokemonsResponse {
+    count: number;
+    next: string | null;
+    previous: string | null;
+    results: Pokemon[];
+}
+
+const PokemonsList = () => {
+    const [pokemons, setPokemons] = useState<Pokemon[]>([]);
+    const [currentPage, setCurrentPage] = useState<number>(1);
+    const perPage: number = 20;
+
+    useEffect(() => {
+        const fetchPokemons = async () => {
+            const res = await getList(perPage * (currentPage - 1), perPage);
+            const pokemonsWithSprites = await Promise.all(res.data.results.map(async (pokemon) => {
+                const sprite = await getOne(pokemon.name);
+                return {
+                    ...pokemon,
+                    sprites: sprite.data.sprites.front_default
+                }
+            }))
+            setPokemons(pokemonsWithSprites);
+        }
+        fetchPokemons();
+    }, [currentPage]);
+
+    return (
+        <div className={styles.container}>
+            <ul className={styles.list}>
+                {pokemons.map((pokemon) => (
+                    <PokemonCard key={pokemon.name} name={pokemon.name} url={pokemon.sprites}/>

```

Comment:
`url` should get a string, but getting `{ front_default: string }` because type says so even if this field is reassigned.